allenact/base_abstractions/task.py

```diff
@@ -428,10 +434,22 @@ def observation_space(self):
 
     def get_observations(self, **kwargs) -> List[Any]:  #-> Dict[str, Any]:
         # Render all tasks in batch
-        self.tasks[0].env.render()  # assume this is stored locally in the env class
+        self.frames = self.render("rgb")
+        self.depths = self.render("depth")
+        self.segs = self.render("seg")
 
         # return {"batch_observations": [task.get_observations() for task in self.tasks]}
-        return [task.get_observations() for task in self.tasks]
+        return [task.get_observations(
+            frame=self.frames[idx],
+            depth=self.depths[idx],
+            seg=self.segs[idx],
+        ) for idx, task in enumerate(self.tasks)]
+
+    def update_state_views(self):
+        for idx, state_view in enumerate(self.state_views):
+            updated_state_view = self.tasks[0].env.call(state_view)
```

Comment:
I know I introduced this notation, but we might prefer to use the env reference in the task sampler rather than that in task 0? I was not sure of what seems cleaner, but I'm currently inclined for the task sampler reference as the better option. Again, happy to discuss what seems best :)